airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcTargetPosition.java

```diff
@@ -105,10 +109,27 @@ public boolean isEventAheadOffset(final Map<String, String> offset, final Change
 
     final JsonNode offsetJson = Jsons.deserialize((String) offset.values().toArray()[0]);
 
-    final String offset_lsn =
-        offsetJson.get("lsn_commit") != null ? String.valueOf(offsetJson.get("lsn_commit")) : String.valueOf(offsetJson.get("lsn"));
-    final String event_lsn = String.valueOf(event.eventValueAsJson().get("source").get("lsn"));
-    return Long.parseLong(event_lsn) > Long.parseLong(offset_lsn);
+    final String stateOffsetLsnCommit = offsetJson.get("lsn_commit") != null ? String.valueOf(offsetJson.get("lsn_commit")) : null;
+    if (stateOffsetLsnCommit == null) {
+      return false;
+    }
+
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      TypeReference<List<String>> listType = new TypeReference<>() {};
+      final JsonNode lsnSequenceNode = event.eventValueAsJson().get("source").get("sequence");
```

Comment:
Is there a case sequence isn't going to be there?

Reply:
I don't think so. It's been there since at least 2.2, possibly longer : https://debezium.io/documentation/reference/2.2/connectors/postgresql.html#postgresql-create-events

Reply:
looks like autoformat kicked in. Let's disable it, as your formatting was way more readable

Reply:
Nice seems like that did the trick